pkg/fqdn/matchpattern/matchpattern_test.go

```diff
@@ -27,10 +27,13 @@ var _ = Suite(&MatchPatternTestSuite{})
 // *cilium.io. -> "([a-zA-Z0-9]+[.])?cilium[.]io[.]
 func (ts *MatchPatternTestSuite) TestAnchoredMatchPatternREConversion(c *C) {
 	for source, target := range map[string]string{
-		"cilium.io.":   "^cilium[.]io[.]$",
-		"*.cilium.io.": "^" + allowedDNSCharsREGroup + "*[.]cilium[.]io[.]$",
-		"*":            "(^(" + allowedDNSCharsREGroup + "+[.])+$)|(^[.]$)",
-		".":            "^[.]$",
+		"cilium.io.":    "^cilium[.]io[.]$",
+		"*.cilium.io.":  "^" + allowedDNSCharsREGroup + "*[.]cilium[.]io[.]$",
+		"**.cilium.io.": "^(" + allowedDNSCharsREGroup + "+[.])+" + "cilium[.]io[.]$",
+		"_sub.**.io.":   "^_sub[.](" + allowedDNSCharsREGroup + "+[.])+io[.]$",
+		"**":            "(^(" + allowedDNSCharsREGroup + "+[.])+$)|(^[.]$)",
+		"*":             "(^(" + allowedDNSCharsREGroup + "+[.])+$)|(^[.]$)",
+		".":             "^[.]$",
```

Comment:
some more ideas for test cases (admittedly only some of these related to the PR)

```
"**."
"*."
"**.cilium.**.io"
"**cilium.**.io"
"cilium.**.**.io"
"**.**.cilium.io"
"**.**."
"**.**"
"cilium.**.**"
"cilium.**.**"
```

Reply:
Thank you for suggestions!